game.c

```diff
@@ -105,27 +132,23 @@ int move_player() {
             return -1;
     }
 
-    if (is_position_valid(x, y)) {
-        player.x = x;
-        player.y = y;
-    }
+    move_if_valid(&player, x, y);
     return 0;
 }
 
-void move_dood(struct dood *d) {
-    int x = d->x;
-    int y = d->y;
-    switch (d->behavior) {
+void move_mobile(struct mobile *mob) {
+    int x = mob->x;
+    int y = mob->y;
+    switch (mob->behavior) {
         case RandomWalk:
             if (rand()%2 == 0) {
                 x += rand()%3 - 1;
             } else {
                 y += rand()%3 - 1;
             }
     }
-    if (is_position_valid(x, y)) {
-        d->x = x;
-        d->y = y;
+    if (!(move_if_valid(mob, x, y))) {
+        mob->emote = OUCH;
```

Comment:
I like this. It's fun. I don't actually think I've seen this in rougelikes before.

Reply:
Originally I wanted to print "ouch" but then realized that would require a lot of change to the drawing routine.